### PR TITLE
Add missing stuff to the converters

### PR DESCRIPTION
**Pull request by Disper**
**Description**

Changes proposed in this pull request:

- Fixes for Openstack provider specific config
- Fixes for handling cloud profile name
- Handling default machine image version added
- Refactoring:
  - Networking extender inlined into converter
  - Kubernetes version setting, and OIDC config split 

**Related issue(s)**

### CODE REVIEW

File: internal/gardener/shoot/hyperscaler/openstack/config.go [hunk 1]
```diff
@@ -8,25 +8,31 @@ import (
...
 }
 
-func NewInfrastructureConfig() v1alpha1.InfrastructureConfig {
+func NewInfrastructureConfig(workerCIDR string) v1alpha1.InfrastructureConfig {
 	return v1alpha1.InfrastructureConfig{
 		TypeMeta: v1.TypeMeta{
 			Kind:       infrastructureConfigKind,
 			APIVersion: apiVersion,
 		},
+		FloatingPoolName: defaultFloatingPoolName,
```
[the remark this answers is not in the document]
Comment (Disper) [reply internal/gardener/shoot/hyperscaler/openstack/config.go:32]: Yes, will create a separate issue.

File: internal/gardener/shoot/extender/kubernetes_version_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		kubernetesVersionExtender := NewKubernetesVersionExtender("1.99")
+		err := kubernetesVersionExtender(runtime, &shoot)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, "1.88", shoot.Spec.Kubernetes.Version)
+	})
+}
+
+func ToPtr[T any](v T) *T {
```
Comment (Disper) [inline internal/gardener/shoot/extender/kubernetes_version_test.go:49]: can we move it to some util?

File: internal/gardener/shoot/extender/provider.go [hunk 3]
```diff
@@ -102,3 +95,40 @@ func getZones(workers []gardener.Worker) []string {
 
 	return slices.Compact(zones)
 }
+
+func setWorkerConfig(provider *gardener.Provider, providerType string, enableIMDSv2 bool) error {
+	if providerType != hyperscaler.TypeAWS || !enableIMDSv2 {
```
[the remark this answers is not in the document]
Comment (Disper) [reply internal/gardener/shoot/extender/provider.go:100]: discussed, we will keep it as it is

**Review by Disper — COMMENTED** [review]
LGTM

**Review by akgalwas — APPROVED** [review]

**Pull request merged**